source/server/connection_handler_impl.cc

```diff
@@ -93,7 +93,10 @@ void ConnectionHandlerImpl::removeFilterChains(
       return;
     }
   }
-  NOT_REACHED_GCOVR_EXCL_LINE;
+  // Reach here if the target listener was removed by a full listener update.
+  // Completion must be deferred called so that the deferred deleting connection can refer the
+  // filter chains.
```

Comment:
"The completion must be deferred so that any active connection referencing the filter chain can finish prior to deletion."?